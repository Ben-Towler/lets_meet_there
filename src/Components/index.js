import SiteHeader from './SiteHeader/SiteHeader.component';
import SearchForm from './SearchForm/SearchForm.component';
import Trip from './Trip/Trip.component';
import Button from './Button/Button.component';
import Form from './Form/Form.component';
import Flight from './Flight/Flight.component';
import FlightSummary from './FlightSummary/FlightSummary.component';
import BookingLink from './BookingLink/BookingLink.component';
import Spinner from './Spinner/Spinner.component';
import LocationSearch from './LocationSearch/locationSearch.component.tsx';

export {
  SiteHeader,
  BookingLink,
  SearchForm,
  Trip,
  Flight,
  FlightSummary,
  Button,
  Form,
  Spinner,
  LocationSearch
}