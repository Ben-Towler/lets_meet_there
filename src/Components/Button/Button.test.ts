import React from 'react';
import renderer from 'react-test-renderer';
import {Button} from 'Components';
import { shallow, configure } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
configure({adapter: new Adapter()});

describe('Booking Link', () => {

  test('snapshot renders', () => {
    const component = renderer.create(
    <Button 
      type='submit'
      children='Submit'
    />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it calls func on button click', () => {
    const mockFunction = jest.fn();
  
    const component = shallow (
      <Button 
        type='submit'
        onClick={mockFunction}
        children='Submit'
      />
    );
    
    component.simulate('click');
    expect(mockFunction).toHaveBeenCalled();
  });
});