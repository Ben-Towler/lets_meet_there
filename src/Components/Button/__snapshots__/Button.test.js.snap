// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Booking Link snapshot renders 1`] = `
<button
  className="sc-AxjAm fDEmJb"
  type="submit"
>
  Submit
</button>
`;
